### PR TITLE
[ADD] fieldservice_agreement_helpdesk

### PR DESCRIPTION
**Pull request by osi-scampbell**
This module links Field Service, Agreements, and Helpdesk together.

### CODE REVIEW

**Review by max3903 — APPROVED** [review]
LGTM

File: fieldservice_agreement_helpdesk/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    "depends": [
+        "helpdesk_fieldservice",
+        "agreement_helpdesk",
+        "fieldservice_agreement"
+    ],
+    "data": [
+        "views/helpdesk_ticket.xml",
+        "views/fsm_order.xml",
+    ],
+    "installable": True,
```
Comment (max3903) [inline fieldservice_agreement_helpdesk/__manifest__.py:20]:
```suggestion
    "installable": True,
    "auto_install": True,
```

File: fieldservice_agreement_helpdesk/views/fsm_order.xml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+<odoo>
+
+    <!-- FieldService Order Form View -->
+    <record id="view_fsm_order_ticket_form" model="ir.ui.view">
+        <field name="name">view.fsm.order.ticket.form</field>
+        <field name="model">fsm.order</field>
+        <field name="inherit_id" ref="fieldservice.fsm_order_form"/>
+        <field name="arch" type="xml">
+            <field name="agreement_id" position="attributes">
+                <attribute name="context">{'default_agreement_id': agreement_id}</attribute>
```
Comment (max3903) [inline fieldservice_agreement_helpdesk/views/fsm_order.xml]: @osi-scampbell There is no agreement_id field on agreement. You probably want to set the customer field of the agreement:

```
<attribute name="context">{'default_partner_id': partner_id}</attribute>
```

No need for the helpdesk dependency for this. So it should be moved to fieldservice_agreement.

File: fieldservice_agreement_helpdesk/views/fsm_order.xml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+            </field>
+            <field name="serviceprofile_id" position="attributes">
+                <attribute name="context">{'default_serviceprofile_id': serviceprofile_id}</attribute>
```
Comment (max3903) [inline fieldservice_agreement_helpdesk/views/fsm_order.xml]:
```suggestion
                <attribute name="context">{'default_agreement_id': agreement_id}</attribute>
```

No need for the helpdesk dependency for this. So it should be moved to fieldservice_agreement.

**Review by max3903 — CHANGES_REQUESTED** [review]
Minor changes.

File: fieldservice_agreement_helpdesk/README.rst [hunk 4]
```diff
@@ -0,0 +1,28 @@
+.. image:: https://img.shields.io/badge/licence-LGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/lgpl-3.0-standalone.html
+   :alt: License: LGPL-3
+
+====================================
+Field Service - Agreement - Helpdesk
+====================================
+
+This module links Field Service, Agreements, and Helpdesk together.
```
Comment (max3903) [inline fieldservice_agreement_helpdesk/README.rst]:
```suggestion
This module links Field Service, Agreements and Help Desk together by:

* setting the agreement fields when creating Field Service orders from a ticket.
```

File: fieldservice_agreement_helpdesk/views/fsm_order.xml (hunk 3 above)
Comment (max3903) [reply fieldservice_agreement_helpdesk/views/fsm_order.xml]: Ping @osi-scampbell

Comment (osi-scampbell) [reply fieldservice_agreement_helpdesk/views/fsm_order.xml]: Removed fsm_order.xml. Will make changes under fieldservice_agreement.

**Review by max3903 — APPROVED** [review]

**Pull request merged**